config/core/configmaps/defaults.yaml

```diff
@@ -108,3 +108,15 @@ data:
     # allow-container-concurrency-zero controls whether users can
     # specify 0 (i.e. unbounded) for containerConcurrency.
     allow-container-concurrency-zero: "true"
+
+    # Whether to disable service links on knative services.
+    # If set to `true` the service links will be disabled for all
+    # knative services.
+    # If set to `false` kubernetes default (true) or explicitly
+    # specified value will be propagated.
+    # See: https://kubernetes.io/docs/concepts/services-networking/connect-applications-service/#accessing-the-service
+    #
+    # In environments with large number of services it is suggeted
+    # to be set to `false`.
+    # See https://github.com/knative/serving/issues/8498.
+    disable-service-links: "false"
```

Comment:
I think I'd be more comfortable orienting this to be the same as Kubernetes' field and simple making this a configurable default thereof.

Reply:
Originally I wrote it this way. But it doesn't permit per service settings, which I think we should permit.
The alternative is tri-state flag.